.github/copilot-instructions.md

```diff
@@ -69,65 +69,76 @@ pixi run -e pbuild build-wheels
 
-## 7. Testing Strategy
+## 8. Testing Strategy
 - Unit tests target granular modules (`tests/python/unit/...`). Add new tests adjacent to similar domain (e.g., new utility → `tests/python/unit/utilities/`).
 - Integration tests at `tests/python/integration` cover full pipelines.
 - Coverage thresholds enforced (`--cov-fail-under=30` for unit suite). Keep defensive code minimal; exclude per coverage config if necessary.
 - Rust tests live in crate `src` using standard Cargo conventions; prefer small, deterministic tests.
 
-## 8. Logging & Observability
+## 9. Logging & Observability
 - Queue-based logging ensures non-blocking writes; to add new structured exception logging, extend `_JsonExceptionFileHandler` or filters.
 - Location-specific run logs + JSON exception summaries stored under analysis run directories.
 - Use `COMPASS_DEBUG_LEVEL` env if needed for custom debug filtering (present pattern). Custom log levels defined: `TRACE=5`, `DEBUG_TO_FILE=9`.
 
-## 9. Common Pitfalls & Gotchas
+## 10. Common Pitfalls & Gotchas
 - Playwright version range pinned; mismatched versions can break stealth/c4ai features. Use Pixi lock or `--frozen` installation.
 - Do not modify `_version.py` directly; tests or builds may assume dynamic version structure.
 - Browser install required for crawl; forgetting `playwright install` causes runtime failures.
 - Some services rely on external API keys (OpenAI, Anthropic); code should gracefully handle missing keys (check provider module patterns).
-- When adding new dependencies ensure compatibility with Ruff selected rules; if formatting conflicts, examine ignore list before adding new ignores.
 
-## 10. Implementing a Sample Feature (Reference Flow)
+## 11. Implementing a Sample Feature (Reference Flow)
 Example: Add a helper to print current package version.
 1. Import `from compass import __version__`.
 2. Provide CLI subcommand or simple script; place in `compass/scripts/` or extend `_cli/main.py`.
 3. Write unit test in `tests/python/unit/` asserting semantic version pattern.
 4. Run `pixi run -e pdev tests-u` → refine on failures.
 
-## 11. CI/Release Lifecycle
+## 12. CI/Release Lifecycle
 - PR: Ruff lint → unit tests (locked + multi-OS) → integration tests. Fix lint first to avoid wasted CI cycles.
 - Rust: check → fmt/clippy → tests.
 - Docs must build with `--fail-on-warning`; ensure new references are valid.
 
-## 12. Contribution & Style Extensions
-- Follow Development Guidelines in published docs for deeper details.
+## 13. Contribution & Style Extensions
+- Follow Development Guidelines in `docs/source/dev/README.rst` for deeper details.
 - Keep functions small; prefer pure helpers in `utilities/` with focused responsibilities.
-- Prefer dataclass or simple dicts over complex inheritance unless mirroring existing pattern in `services/` or `extraction/`.
+- Prefer simple dicts over dataclasses.
```

Comment:
I'm curious to see if the `prefer` will work well on weighting the preference.

Reply:
We can monitor this!